core/src/main/java/com/graphhopper/util/exceptions/ConnectionNotFoundException.java

```diff
@@ -28,4 +29,29 @@
     public ConnectionNotFoundException(String var1, Map<String, Object> details) {
         super(var1, details);
     }
+
+    /**
+     * You can add a reason for this exception, to add more details why this happened.
+     */
+    public ConnectionNotFoundException(String var1, Map<String, Object> details, Reason reason) {
+        super(var1, details);
+
+        if (details == null) {
+            details = new HashMap<>();
+        }
+        details.put("reason", reason.toString());
+    }
+
+    public enum Reason {
```

Comment:
Why a separate class?

Reply:
Yes maybe I over engineered here a bit. I think my idea was that we could have different reasons and this would make it easier always return the same reason from different locations. Should I just make it a String?

Reply:
Yes, for now a string seems good. Is the Map as parameter in the constructor the same way like we do it for other exceptions? Or should we make the map internal & final and allow changing it via `exception.set(key, value)`?

Reply:
Ok let's change it to a String :).